octoprint_SpoolManager/static/js/SpoolManager.js

```diff
@@ -299,12 +299,11 @@ $(function() {
                 ){
                 var check = confirm('External database will not work. Save settings anyway?');
                 if (check == true) {
-                    origSaveSettingsFunction(data, successCallback, setAsSending);
+                    return origSaveSettingsFunction(data, successCallback, setAsSending);
                 }
-                return
+                return null;
```

Comment:
OctoPrint's version of this always returns the promise from `OctoPrint.settings.save()`. So while this fixes a lot of the issues, it does not fix all of them because the signature is still different.

Must find a way to always return the settings promise from here, regardless of whether you want it to or not. If monkey patching is going to exist (which, again it should not) then it must be done perfectly, to behave as close to the original function.